File: frontend/apps/admin/features/fa-im-pages/types/friend/ImFriend.ts
// types/friend/ImFriend.ts
export interface ImFriend {
  id: number;              // 主键ID
  userId: number;          // 用户ID
  friendId: number;        // 好友ID
  friendGroupId?: number;  // 好友分组ID
  remark?: string;         // 好友备注
  deleted?: boolean;       // 删除状态
  crtTime?: string;        // 创建时间
  updTime?: string;        // 更新时间
}

File: fa-im/src/main/java/com/faber/api/im/friend/entity/ImFriend.java
package com.faber.api.im.friend.entity;

import com.faber.core.bean.BaseDelEntity;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 用户好友关系实体类
 *
 * @author faber
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("im_friend")
public class ImFriend extends BaseDelEntity {
    /**
     * 用户ID
     */
    @TableField("user_id")
    private Long userId;

    /**
     * 好友ID
     */
    @TableField("friend_id")
    private Long friendId;

    /**
     * 好友分组ID
     */
    @TableField("friend_group_id")
    private Long friendGroupId;

    /**
     * 好友备注
     */
    @TableField("remark")
    private String remark;
}